vaadin-grid-flow-demo/src/main/java/com/vaadin/flow/component/grid/demo/GridDemo.java

```diff
@@ -2511,8 +2509,11 @@ private void createDragData() {
                         ? Collections.max(persons.stream().collect(Collectors
                                 .mapping(Person::getId, Collectors.toList())))
                         : 0;
-                persons.add(index, new Person(maxId + 1, dataArray[0],
-                        dataArray[1], -1, null, dataArray[2]));
+                if (dataArray.length == 3) {
```

Comment:
I think we can make it in a better way. For example, we can make this change in a way that it does not accept copied data from other demos. I suggest forcing this demo not to accept the data from other demos data by adding this code in addDropListener like this:

            if(event.getSource() != grid)
                return;

Now the code will look like this:
        grid.addDropListener(event -> {
            if(event.getSource() != grid)
                return;

            int index = event.getDropTargetItem().map(person -> persons
                    .indexOf(person)
                    + (event.getDropLocation() == GridDropLocation.BELOW ? 1
                            : 0))
                    .orElse(0);

            String personsData = event.getDataTransferText();
            Arrays.asList(personsData.split(";")).forEach(personData -> {
                String[] dataArray = personData.split(",");
                Integer maxId = !persons.isEmpty()
                        ? Collections.max(persons.stream().collect(Collectors
                                .mapping(Person::getId, Collectors.toList())))
                        : 0;
                persons.add(index, new Person(maxId + 1, dataArray[0],
                        dataArray[1], -1, null, dataArray[2]));
            });
            grid.getDataProvider().refreshAll();
        });

I think this would be easier to understand for our users.

Reply:
This is true and I think I tried it, but `event.getSource` returned grid for me anyway....

Reply:
Yes, unfrotuantely, it would not work as it always returns `Grid` (It returns the component whose Event it is, not the one who triggers it. Which is indeed misleading) 
https://vaadin.com/api/platform/14.2.0/com/vaadin/flow/component/grid/Grid.html#addDropListener-com.vaadin.flow.component.ComponentEventListener-

Reply:
@shadikhani

Reply:
Yes, you are right!